src/optimizer.cpp

```diff
@@ -1290,6 +1295,11 @@ void Optimizer::optimize_integer_constraints(Hierarchy& mRes, std::map<int, int>
             if (flow_count == supply) fullFlow = true;
             if (level != 0 || fullFlow) break;
             edge_capacity += 1;
+            iter++;
+            if (iter == 10) {
```

Comment:
I tried to figure out why this would get stuck without coming to a solution, but couldn't really find anything. So I instead introduced a maximum number of tries until we should simply give up.